### ISSUE TITLE
hackathon?

### ISSUE THREAD

**Issue by bbolker** [glmmTMB/glmmTMB#976]
Floating the idea of a `glmmTMB` hackathon sometime this year (maybe in the spring/summertime)?  I've been chipping away at things but there's still quite a bit of low-hanging fruit and it would be fun/interesting to try to regain some momentum. 

* The very easy way to do this would be remotely via Zoom/Discord/etc.
* if people were feeling ambitious and could get travel money, we could almost certainly get support for a [focused research group at the Banff International Research Station](https://www.birs.ca/programs/general-program-descriptions#Focused-Research-Groups) (there's a 4-month lead time).

**Comment by mebrooks** [comment]
Yes, I agree it would be nice to focus on this for a few days or a week. If many developers are planning to attend ISEC (15-19 July 2024 Swansea, UK) maybe we could plan something before or after that, nearby. I'd also be happy to come to Banff, but need to find travel money. Somewhere, I have notes on some funding options, but I'll have to dig them out.

**Comment by MrJerryTAO** [comment]
I am interested in this.

**Comment by bbolker** [comment]
I'm seriously considering ISEC, and would be happy to stay a day or so afterward to hack. (I won't be available before.) Perhaps we could poll developers/other interested people?  Could we contact the ISEC organizers and see if we might be able to borrow some working space for a couple of days?

**Comment by mebrooks** [comment]
Just an idea... if we had one hackathon in Denmark, maybe we could involve @gavinsimpson and tackle some [GAMM plotting methods ](https://github.com/gavinsimpson/gratia/issues/240).

**Comment by gavinsimpson** [comment]
I'm 95% sure that I'll be in Swansea for ISEC, but will have to leave right away after the conference, unfortunately.

As for Denmark, I'd be happy to host something in Aarhus, although my availability would depend on the exact timing.

**Comment by bbolker** [comment]
Time to think about this some more.

- I will be at ISEC but *not* available beforehand.
- I'm happy to come to Copenhagen or Aarhus for a day after ISEC if that works well for people.
- While we'd like to have the largest possible core group in person, I'm sure we could work out some hybrid options too.

I started a Google sheet [here](https://docs.google.com/spreadsheets/d/1N7lFrMfxKmYq5KVfFdkQUa4VfRqX6MK-I2INwgM0dJI/edit?usp=sharing) so who wants to/can participate.

Once we know where/when we want to do this we need to find a venue (we probably just need a room for a day), unless we have to fall back to a virtual-only event (in which case our time constraints get much looser)

**Comment by mebrooks** [comment]
I can't find any funding to host a meeting at the moment. Of course, I'm sure we can use space at DTU. A meeting starting July 22 in Copenhagen seems like it might be possible.

I noticed that ISEC ends at 13:00 on Friday the 19th of July, so maybe we could do a half-day meeting there. The registration system seems to still be offering dorm rooms that evening. This is assuming that people haven't already booked their flights.

**Comment by bbolker** [comment]
If people are still here/paying attention ... if it's still a possibility, I think a Friday afternoon half-day makes a lot of sense -- certainly it's the lowest-hanging fruit.  We could make it a long session (pizza etc.) if people had the energy.  *If* people would still be around and would be able to also do Saturday morning, I bet we could ask Luca Borger for a room somewhere ... 

(I am being lazy and this feels easier to me than schlepping to Denmark, although that would be fun.)

Depending on our priorities we could also make it a more open meeting/feedback session - we would get less done but we might be able to hear from a broader audience about priorities.

I am also going to send an e-mail about this to the people who listed themselves on the Google sheet.

**Comment by mebrooks** [comment]
Sounds perfect to me as I might need to move to a new apartment the weekend before the conference.

In the future, I'll try to get funding for an in-person developers' meeting.

**Comment by jaganmn** [comment]
I can join if it's hybrid, but if that would be a headache and if most people are joining in person from ISEC, then I can sit this one out.

**Comment by bbolker** [comment]
I suspect we will try for an actively hybrid meeting ...

**Comment by gavinsimpson** [comment]
I'm afraid I won't now be able to make this; I'll be at ISEC but only arrive on the Sunday and will need to leave Friday morning so as to pick the kids up from my parents'.

**Comment by bbolker** [comment]
I asked Luca Borger about finding a room. [This](https://www.canva.com/design/DAFYMnFB_QQ/6qJjp9r-a0j4chrsFBlKig/view#2) might also be a possibility ...

**Comment by mebrooks** [comment]
I'm booking my flight to return on Saturday evening. So I'll be around Friday after the conference and Saturday morning. I booked Friday dinner in the ISEC accommodation, but am flexible.  I bet we could use some space at the conference center for free.

**Comment by bbolker** [comment]
Summary of yesterday's discussion (from what I can recall):

* we will merge Maeve's `propto` branch ([here](https://github.com/mmcgillycuddy/glmmTMB/tree/prop2_covstruc), I think?) as soon as we can; MM will start a PR and add a checklist of items to do before merging (e.g. merge master back and deal with any conflicts; testing; document downstream issues that are known to break; etc.)
* we'll work to add info on the indices of `beta` and `b` values corresponding to each smooth term, to enable `gratia` functionality
* maybe spend some time thinking/discussing how RE formulae (in the `reformulas` package) should be extended to handle more interesting cases/models requiring auxiliary data. Worth looking at (1) `brms` documentation ([here](http://paul-buerkner.github.io/brms/reference/brmsformula-helpers.html), [here](http://paul-buerkner.github.io/brms/reference/brmsformula.html) and Tanaka and Hui (2019):

Tanaka, Emi, and Francis K. C. Hui. 2019. “Symbolic Formulae for Linear Mixed Models.” In Statistics and Data Science, edited by Hien Nguyen, 3–21. Communications in Computer and Information Science. Singapore: Springer. https://doi.org/10.1007/978-981-15-1960-4_1.  Also http://arxiv.org/abs/1911.08628



Zoom link for remote participants: https://mcmaster.zoom.us/j/92311975707

**Comment by bbolker** [comment]
We'll be in room 022 (or on zoom) at 1330 UK time.  For efficiency, please edit/add to the list below of which issues you think you primarily want to work on :

- [ ] enhance `$modelInfo` (and maybe `getME()`) with information about the assignment of `beta` and `b` elements to particular random effects/smooths, for `gratia` compatibility (**Ben**)
- [ ] work on pull request/clean up for current implementation of `propto()` (**Maeve**)

**Comment by jaganmn** [comment]
(I need a meeting passcode.)

**Comment by bbolker** [comment]
1 minute

**Comment by mebrooks** [comment]
I just noticed that Banff has [an affiliate in Mexico](https://www.birs.ca/cmo). This could come in handy since the next ISEC will be in Mexico in Jan 2027.  However, it looks like it might be closed December to March according to [the guidelines](https://proposals.birs.ca/guidelines).
